homeassistant/components/mqtt/device_automation.py

```diff
@@ -0,0 +1,46 @@
+"""Provides device automations for MQTT."""
+import logging
+
+import voluptuous as vol
+
+from homeassistant.components import mqtt
+from homeassistant.helpers.dispatcher import async_dispatcher_connect
+
+from . import ATTR_DISCOVERY_HASH, device_trigger
+from .discovery import MQTT_DISCOVERY_NEW, clear_discovery_hash
+
+_LOGGER = logging.getLogger(__name__)
+
+AUTOMATION_TYPE_TRIGGER = "trigger"
+AUTOMATION_TYPES = [AUTOMATION_TYPE_TRIGGER]
+AUTOMATION_TYPES_SCHEMA = vol.In(AUTOMATION_TYPES)
+CONF_AUTOMATION_TYPE = "automation_type"
+
+PLATFORM_SCHEMA = mqtt.MQTT_BASE_PLATFORM_SCHEMA.extend(
+    {vol.Required(CONF_AUTOMATION_TYPE): AUTOMATION_TYPES_SCHEMA},
+    extra=vol.ALLOW_EXTRA,
+)
+
+DEVICE_TRIGGERS = "mqtt_device_triggers"
```

Comment:
This is defined twice. Also in device_trigger.py

Reply:
Removed from here.